tidy3d/components/simulation.py

```diff
@@ -831,6 +1055,39 @@ def monitor_medium(self, monitor: MonitorType):
         medium = medium_set.pop() if medium_set else self.medium
         return medium
 
+    @staticmethod
+    def _check_bloch_vec(
+        source: SourceType,
+        bloch_vec: float,
+        dim: Axis,
+        medium: MediumType,
+        domain_size: float,
+    ):
+        """Helper to check if a given Bloch vector is consistent with a given source."""
+
+        # make a dummy Bloch boundary to check for correctness
+        dummy_bnd = BlochBoundary.from_source(
+            source=source, domain_size=domain_size, axis=dim, medium=medium
+        )
+        expected_bloch_vec = dummy_bnd.bloch_vec
+        if bloch_vec != expected_bloch_vec:
+            test_val = np.real(expected_bloch_vec - bloch_vec)
+            if test_val.is_integer() and test_val != 0:
+                log.warning(
+                    f"The wave vector of source '{source.name}' along dimension "
+                    f"{dim} is equal to the Bloch vector of the simulation "
```

Comment:
double check the `'` quotes around some things in here, ie `{dim}`, `DiffractionMonitor`, `BlochBoundary.from_source`. and the SetupError below